src/woodbury_pd_mat.jl

```diff
@@ -88,3 +88,7 @@ end
 # NOTE: the parameterisation to scale up the Woodbury matrix is not unique. Here we
 # implement one way to scale it.
 *(a::WoodburyPDMat, c::Real) = WoodburyPDMat(a.A, a.D * c, a.S * c)
+*(c::Real, a::WoodburyPDMat) = a * c
+*(a::WoodburyPDMat, c::Diagonal{T}) where {T<:Real} = WoodburyPDMat(sqrt(c) * a.A, a.D, a.S * c)
+*(c::Diagonal{T}, a::WoodburyPDMat) where {T<:Real} = a * c
+*(c1::Diagonal{T}, a::WoodburyPDMat, c2::Diagonal{T}) where {T<:Real} = WoodburyPDMat(sqrt(c1) * sqrt(c2) * a.A, a.D, c1 * a.S * c2)
```

Comment:
same concern on c1/c2 has negative values. Why we don't adjust `D`?

Reply:
same comment as above